docs/enhancements/0-single-daemonset.md

```diff
@@ -0,0 +1,278 @@
+# Single DaemonSet for KMM
+
+Authors: @qbarrand, @mresvanis, @yevgeny-shnaidman
+
+## Introduction
+
+This enhancement aims at eliminating several pain points related to how KMM loads kernel modules on the nodes.
+
+### Resource utilization
+
+KMM currently creates one DaemonSet per `Module` and compatible kernel version found in the cluster.
+Formally, the number of DaemonSets created $S$ is defined by:
+$$S = \sum_{i=1}^nk_i$$
+
+where:
+- $n$ is the number of `Modules` in the cluster, and;
+- $k_i$ is the number of kernel versions running in the cluster that `Module` $i$ is compatible with.
+
+A big $S$ leads to unnecessary workload for the cluster:
+- on nodes, one ModuleLoader Pod is created per `Module`. Although those just sleep after the `modprobe`
+  `postStart` hook, each running pod incurs an overhead for the container runtime and the kubelet;
+- on the control-plane, a higher number of DaemonSets and Pods creates more events for kube-controller-manager, which
+  trigger more reconciliations and status updates.
+
+### Reliability
+
+Running user-provided ModuleLoader images as DaemonSets makes it difficult for KMM to determine if a module was
+successfully unloaded or not.  
+KMM v1 uses the `preStop` hook to run `modprobe -r` and unload modules.
+`preStop` hook failures cannot prevent the Pod from being terminated.
+If `modprobe` cannot to unload the module for any reason (e.g. userspace applications still using it), then the Pod will
+be terminated anyway and the `kmm.node.kubernetes.io/$namespace.$module.ready` label will be removed from the Node, even
+though the module is still loaded.
+
+## Goals
+
+1. Reduce the overall KMM footprint
+2. Improve the reliability of module loading and unloading
+
+## Non-goals
+
+1. Add further control on the deployment of kmods on nodes
+
+## Design
+
+### KMM: one DaemonSet pulling all kmod images
+
+This proposal reduces the number of DaemonSets run by KMM to one.
+KMM is now composed of two components, communicating over the network:
+- the control-plane is the current operator Deployment;
```

Comment:
I assume that the `control-plane` component doesn't have to run on the master nodes right?! In this case, `control-plane` is a bit confusing. One can think we have a similar type of deployment as NFD does where some of the pods runs on master nodes.